app/assets/javascripts/geoblacklight/viewers/viewer.js

```diff
@@ -8,7 +12,44 @@ GeoBlacklight.Viewer = L.Class.extend({
 
     L.Util.setOptions(this, options);
 
-    // trigger viewer load function
+    // trigger viewer load functions
     this.load();
+  },
+
+  /**
+  * Loads leaflet controls from controls directory.
+  **/
+  loadControls: function() {
+    var _this = this;
+    var protocol = this.data.protocol.toUpperCase();
+    var options = this.data.leafletOptions;
+
+    if (!options.VIEWERS) {
+      return;
+    }
+
+    var viewer = options.VIEWERS[protocol];
+    var controls = viewer && viewer.CONTROLS;
+
+    this.controlPreload();
+
+    /**
+    * Loop though the GeoBlacklight.Controls hash, and for each control,
+    * check to see if it is included in the controls list for the current
+    * viewer. If it is, then pass in the viewer object and run the function
+    * that adds it to the map.
+    **/
+    $.each(GeoBlacklight.Controls, function(name, func) {
+      if (controls && controls.indexOf(name) > -1) {
+        func(_this);
```

Comment:
It would probably be more javascript-y to explicitly set the context on the function call:

`func.call(_this)`

which would mean the viewer becomes `this` in a control function.